src/components/SeatMap/SeatMap.test.js

```diff
@@ -0,0 +1,554 @@
+      const seatMapDataForOneDeck = {
+        params: paramsData(),
+        content: [
+          finalDeck({
+            rows: [
+              row({
+                seats: [
+                  finalSeat({
+                    number: '11A',
+                  }),
+                ],
+              }),
+            ],
+          }),
+        ],
+        exits: [[]],
+        bulks: [[]],
+      };
+      JetsSeatMapService.prototype.getSeatMapData = jest.fn().mockResolvedValue(seatMapDataForOneDeck);
+
+      setup({
+        flight: flightDetails(),
+        availability: null,
+        passengers: null,
+        currentDeckIndex: 1,
+        configOverrides: { visibleFuselage: false },
+      });
+
+      await waitFor(() => {
+        expect(screen.queryByTestId('jets-nose')).not.toBeInTheDocument();
+        expect(screen.queryByTestId('jets-tail')).not.toBeInTheDocument();
+      });
+    });
+  });
+
+  describe('when the seat map is called with valid flight data', () => {
+    it('should render seat for a single-deck configuration', async () => {
+      const seatMapDataForOneDeck = {
+        params: paramsData(),
+        content: [
+          finalDeck({
+            rows: [
+              row({
+                seats: [
+                  finalSeat({
+                    number: '11A',
+                  }),
+                ],
+              }),
+            ],
+          }),
+        ],
+        exits: [[]],
+        bulks: [[]],
+      };
+
+      JetsSeatMapService.prototype.getSeatMapData = jest.fn().mockResolvedValue(seatMapDataForOneDeck);
+
+      setup({
+        flight: flightDetails(),
+        availability: null,
+        passengers: null,
+        currentDeckIndex: 0, // lower deck
+      });
+
+      await waitFor(() => {
+        const seatNumber = screen.queryByText(/11A/);
+        expect(seatNumber).toBeInTheDocument();
+        expect(screen.queryByTestId('jets-bulk')).not.toBeInTheDocument();
+        expect(screen.queryByTestId('jets-exit')).not.toBeInTheDocument();
+      });
+    });
+
+    it('should render seat availability for a single-deck configuration', async () => {
+      const seatMapDataForOneDeck = {
+        params: paramsData(),
+        content: [
+          finalDeck({
+            rows: [
+              row({
+                seats: [
+                  finalSeat({
+                    number: '11A',
+                    status: 'available',
+                  }),
+                  finalSeat({
+                    number: '11B',
+                    status: 'unavailable',
+                  }),
+                ],
+              }),
+            ],
+          }),
+        ],
+        exits: [[]],
+        bulks: [[]],
+      };
+
+      JetsSeatMapService.prototype.getSeatMapData = jest.fn().mockResolvedValue(seatMapDataForOneDeck);
+
+      setup({
+        flight: flightDetails(),
+        availability: null,
```

Comment:
Although this availability parameter is not being used because getSeatMapData is mocked and the availability returned in the seats is the one defined in the mock, I think it would be helpful to add the fake availability data here that would make seat 11A available, as having this as null in the test is confusing